src/components/ImageSlider2.jsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import {
+  faChevronLeft,
+  faChevronRight,
+} from "@fortawesome/free-solid-svg-icons";
+
+const ImageSlider2 = ({ slides }) => {
+  const [current, setCurrent] = useState(0);
+  const length = slides.length;
+
+  const nextSlide = () => {
+    setCurrent(current === length - 1 ? 0 : current + 1);
+  };
+
+  const prevSlide = () => {
+    setCurrent(current === 0 ? length - 1 : current - 1);
+  };
+
+  if (!Array.isArray(slides) || slides.length <= 0) {
+    return null;
+  }
+
+  return (
+    <section className="font-montserrat relative h-full flex justify-center items-center max-[1023px]:pt-8 ">
+      <FontAwesomeIcon
+        icon={faChevronLeft}
+        size="xl"
+        className="hover:text-[#737373] text-[#FFFFFF] mr-72 absolute"
+        onClick={prevSlide}
+      />
+
+      {slides.map((slide, index) => {
```

Comment:
Burada div içinde yalnızca index === current koşulu sağlandığında içerik gösteriliyor. Bu işlemi daha temiz bir şekilde yönetmek için koşullu renderlamayı map fonksiyonunun dışına çıkarabilirsiniz. Böylece, gereksiz döngü çalıştırmak yerine sadece gerekli resmi renderlamış olursunuz.Örneğin:

```
{slides.length > 0 && (
  <div key={current}>
    <img
      src={slides[current]}
      alt="travel image"
      className="w-full h-full object-cover"
    />
  </div>
)}

```